drivers/modem/gsm_ppp.c

```diff
@@ -741,42 +746,42 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 #endif /* CONFIG_GSM_MUX */
 }
 
+#if IS_ENABLED(CONFIG_GSM_MUX)
 static int mux_enable(struct gsm_modem *gsm)
 {
-	int ret;
+	int ret = 0;
 
 	/* Turn on muxing */
-	if (IS_ENABLED(CONFIG_MODEM_GSM_SIMCOM)) {
```

Comment:
I'm not in favor of changes within this commit - as mentioned earlier, it's easier to introduce some regression if preprocessor checks are used and some particular configuration is not covered by the CI. IMO it's against what we typically do in Zephyr.

Reply:
@rlubos meaning if I do `#if IS_ENABLED(CONFIG_GSM_MUX)` the mux related function will not be checked in CI?

Reply:
It really depends on what samples are actually enabling gsm_ppp module and this particular feature (you'd need to analyze paritcualr samples and their sample.yaml files to see what's actually being built).

Reply:
> It really depends on what samples are actually enabling gsm_ppp module and this particular feature (you'd need to analyze paritcualr samples and their sample.yaml files to see what's actually being built).

That means if we modify the `tests/drivers/build_all/modem`'s testcase.yaml it is possible to test these different features?

Reply:
Build, not test. From what I see, that particular test project seems only to build some certain configurations, it does not have any test cases defined.

Reply:
Is building them good enough?

Reply:
It's the bare minimum I'd say.

Reply:
But then we are compiling hundreds of lines of basically dead codes for users who aren't using the uart_mux and unused modem-specific's mux enabling steps in `mux_enable`

Reply:
Yes, you're right, we're wasting some micro/milliseconds for the extra compilation. But we benefit from better compiler code coverage, and this approach is preferred in Zephyr:
https://docs.zephyrproject.org/latest/reference/util/index.html#c.IS_ENABLED

Reply:
@rlubos got it, thanks!